File: mobile_app/lib/features/shop/domain/shop.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'shop.freezed.dart';
part 'shop.g.dart';

@freezed
abstract class Shop with _$Shop {
  const factory Shop({
    required int id,
    @JsonKey(name: 'owner_id') required int ownerId,
    required String name,
    required String type,
    String? description,
    required String address,
    required String city,
    required String province,
    double? latitude,
    double? longitude,
    List<String>? photos,
    @JsonKey(name: 'business_license') String? businessLicense,
    @JsonKey(name: 'owner_nid') String? ownerNid,
    required String phone,
    String? email,
    @Default('pending') String status,
    @JsonKey(name: 'rejection_reason') String? rejectionReason,
    @JsonKey(name: 'approved_at') DateTime? approvedAt,
    @JsonKey(name: 'primary_photo_url') String? primaryPhotoUrl,
    @JsonKey(name: 'business_license_url') String? businessLicenseUrl,
    @JsonKey(name: 'created_at') DateTime? createdAt,
  }) = _Shop;

  const Shop._();

  bool get isPending => status == 'pending';
  bool get isApproved => status == 'approved';
  bool get isRejected => status == 'rejected';
  bool get isSuspended => status == 'suspended';

  factory Shop.fromJson(Map<String, dynamic> json) => _$ShopFromJson(json);
}

File: backend/app/Models/Shop.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Shop extends Model
{
    protected $fillable = [
        'owner_id',
        'name',
        'type',
        'description',
        'address',
        'city',
        'province',
        'latitude',
        'longitude',
        'photos',
        'business_license',
        'owner_nid',
        'phone',
        'email',
        'status',
        'rejection_reason',
        'approved_at',
    ];

    protected $casts = [
        'photos' => 'array',
        'latitude' => 'double',
        'longitude' => 'double',
        'approved_at' => 'datetime',
    ];

    protected $appends = ['primary_photo_url', 'business_license_url'];

    public function getPrimaryPhotoUrlAttribute()
    {
        if ($this->photos && count($this->photos) > 0) {
            $photo = $this->photos[0];
            if (filter_var($photo, FILTER_VALIDATE_URL)) {
                return $photo;
            }
            return asset('storage/' . $photo);
        }
        return null;
    }

    public function getBusinessLicenseUrlAttribute()
    {
        if ($this->business_license) {
            return asset('storage/' . $this->business_license);
        }
        return null;
    }

    public function owner()
    {
        return $this->belongsTo(User::class, 'owner_id');
    }

    public function products()
    {
        return $this->hasMany(Product::class);
    }

    public function isPending()
    {
        return $this->status === 'pending';
    }

    public function isApproved()
    {
        return $this->status === 'approved';
    }

    public function isRejected()
    {
        return $this->status === 'rejected';
    }

    public function isSuspended()
    {
        return $this->status === 'suspended';
    }
}
